server/src/main/java/org/candlepin/dto/api/v1/ActivationKeyDTO.java

```diff
@@ -0,0 +1,954 @@
+        }
+
+        if (this.products == null) {
+            this.products = new HashSet<String>();
+        }
+
+        return this.products.add(product.getId());
+    }
+
+    /**
+     * Retrieves a view of the content overrides associated with the activation key represented by this DTO.
+     * If the content overrides have not yet been defined, this method returns null.
+     * <p></p>
+     * Note that the collection returned by this method is a view of the collection backing this
+     * set of content overrides. Elements cannot be added to the collection, but elements may be removed.
+     * Changes made to the collection will be reflected by this content overrides data instance.
+     *
+     * @return
+     *  the content overrides associated with this key,
+     *  or null if the content overrideshave not yet been defined
+     */
+    public Set<ActivationKeyContentOverrideDTO> getContentOverrides() {
+        return this.contentOverrides != null ?
+            new SetView<ActivationKeyContentOverrideDTO>(this.contentOverrides) : null;
+    }
+
+    /**
+     * Sets the content overrides of the activation key represented by this DTO.
+     *
+     * @param contentOverrides
+     *  A collection of activation key content override DTOs to attach to this DTO,
+     *  or null to clear the content
+     *
+     * @throws IllegalArgumentException
+     *  if the collection contains null or incomplete content overrides DTOs
+     *
+     * @return
+     *  a reference to this DTO
+     */
+    public ActivationKeyDTO setContentOverrides(Set<ActivationKeyContentOverrideDTO> contentOverrides) {
+        if (contentOverrides != null) {
+            if (this.contentOverrides == null) {
+                this.contentOverrides = new HashSet<ActivationKeyContentOverrideDTO>();
+            }
+            else {
+                this.contentOverrides.clear();
+            }
+
+            for (ActivationKeyContentOverrideDTO dto : contentOverrides) {
+                if (dto == null ||
+                    dto.getContentLabel() == null || dto.getContentLabel().isEmpty() ||
+                    dto.getName() == null || dto.getName().isEmpty() ||
+                    dto.getValue() == null || dto.getValue().isEmpty()) {
+                    throw new IllegalArgumentException("collection contains null or incomplete dtos");
+                }
+
+                this.contentOverrides.add(
+                    new ActivationKeyContentOverrideDTO(
+                    dto.getContentLabel(), dto.getName(), dto.getValue()));
+            }
+        }
+        else {
+            this.contentOverrides = null;
+        }
+        return this;
+    }
+
+    /**
+     * Removes the given content override(s) from this activation key DTO.
+     * @param contentOverride
+     *  The key's associated content override(s) to remove
+     * @return
+     *  True if the content override(s) was removed; false otherwise.
+     */
+    @XmlTransient
```

Comment:
More of a note for myself than for you, but we may want to use JsonIgnore rather than XmlTransient. I'm not remembering which is the deprecated/obsoleted one at present.